### ISSUE TITLE
staticcheck has false positives

### ISSUE THREAD

**Issue by socialsister** [golangci/golangci-lint#4680]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

When I tried to execute `golangci int run -- no configuration -- timeout=10m` for the chainlink project(https://github.com/smartcontractkit/chainlink), the results obtained included:

```go
core/services/keystore/models_test.go:28:2: SA4006: this value of `csa1` is never used (staticcheck)
	csa1, csa2 := csakey.MustNewV2XXXTestingOnly(big.NewInt(1)), csakey.MustNewV2XXXTestingOnly(big.NewInt(2))
	^
core/services/keystore/models_test.go:29:2: SA4006: this value of `eth1` is never used (staticcheck)
	eth1, eth2 := mustNewEthKey(t), mustNewEthKey(t)
	^
core/services/keystore/keys/ocrkey/config_public_key_test.go:32:3: SA4006: this value of `pk` is never used (staticcheck)
		pk := ConfigPublicKey(k.PublicKeyConfig())
```

But in reality, these variables are all used in the following code

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.53.3 built with go1.21rc2 from (unknown, mod sum: "h1:CUcRafczT4t1F+mvdkUm6KuOpxUZTl0yWN/rSU6sSMo=") on (unknown)
```

</details>

### Configuration

<details>

```console
--no-config --timeout=10m   
```

</details>

### Go environment

<details>

```console
$ go version && go env
GO111MODULE='auto'
GOARCH='arm64'
GOBIN=''
GOCACHE='/Users/my/Library/Caches/go-build'
GOENV='/Users/my/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/my/go/pkg/mod'
GONOPROXY=''
GONOSUMDB='
GOOS='darwin'
GOPATH='/Users/my/go'
GOPRIVATE='
GOPROXY='
GOROOT='/Users/my/.g/go'
GOSUMDB='off'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/my/.g/go/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.22.1'
GCCGO='gccgo'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='0'
GOMOD='/Users/my/chainlink/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch arm64 -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/9t/839s3jmj73bcgyp5x_xh3gw00000gn/T/go-build4180674101=/tmp/go-build -gno-record-gcc-switches -fno-common'

```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
core/services/keystore/models_test.go:28:2: SA4006: this value of `csa1` is never used (staticcheck)
	csa1, csa2 := csakey.MustNewV2XXXTestingOnly(big.NewInt(1)), csakey.MustNewV2XXXTestingOnly(big.NewInt(2))
	^
core/services/keystore/models_test.go:29:2: SA4006: this value of `eth1` is never used (staticcheck)
	eth1, eth2 := mustNewEthKey(t), mustNewEthKey(t)
	^
core/services/keystore/keys/ocrkey/config_public_key_test.go:32:3: SA4006: this value of `pk` is never used (staticcheck)
		pk := ConfigPublicKey(k.PublicKeyConfig())
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ldez** [comment]
Hello,

I am not reproducing your problem:

```console
$ git clone org-25111032@github.com:smartcontractkit/chainlink.git
...
$ cd chainlink
$ go mod tidy
$ golangci-lint version                      
golangci-lint has version 1.57.2 built with go1.22.1 from 77a8601a on 2024-03-28T19:01:11Z
$ golangci-lint run --no-config --timeout=10m
$
...
```

I note that you are use an old version of golangci-lint and a non-official build that use go1.21rc2:
```
golangci-lint has version v1.53.3 built with go1.21rc2 from (unknown, mod sum: "h1:CUcRafczT4t1F+mvdkUm6KuOpxUZTl0yWN/rSU6sSMo=") on (unknown)
```

I recommend using the latest stable version of golangci-lint (v1.57.2) and a stable go version (go1.21.9 or go1.22.2)
